packages/core-components/api-report.md

```diff
@@ -2082,6 +2089,18 @@ export type StructuredMetadataTableListClassKey = 'root';
 // @public (undocumented)
 export type StructuredMetadataTableNestedListClassKey = 'root';
 
+// Warning: (ae-forgotten-export) The symbol "SubmenuItemProps" needs to be exported by the entry point index.d.ts
+// Warning: (ae-missing-release-tag) "SubmenuItem" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export const SubmenuItem: ({
+  title,
+  to,
+  hasDropDown,
+  icon: Icon,
+  dropdownItems,
```

Comment:
That is a good point. I did begin considering this while implementing but as this is the first iteration, I opted for the simplest route to get started and making it more flexible in the next iterations. 
The current way allows us to make sure the sidebar item remains active/highlighted when a dropdown item is selected, using the 'to' value. If dropdownItems was JSX, we'd have to consider how we're going to ensure this.

Reply:
We can allow rendering other components by changing `title` in `DropDownItem` type to be JSX and the `to` property could be optional. Would allow the same functionality, just doesn't have to be text.